plumbing/format/packfile/decoder.go

```diff
@@ -201,7 +242,7 @@ func (d *Decoder) DecodeObject() (plumbing.EncodedObject, error) {
 	}
 
 	hash := obj.Hash()
-	d.setOffset(hash, h.Offset)
+	d.setOffsetAndType(hash, h.Offset, obj.Type())
```

Comment:
set... is quite a bad name, as it does not transmit intention or usefulness. Can we call it `memoize` or `remember` instead?